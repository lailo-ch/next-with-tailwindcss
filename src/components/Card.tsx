import * as React from 'react'
import clsx from 'clsx'

export interface Props {
  className?: string | Record<string, unknown>
}

const Card: React.FC<Props> = ({ className, children }) => {
  return (
    <div
      className={clsx(
        'rounded shadow-md',
        'p-4 sm:p-6 md:p-8',
        'bg-white dark:bg-gray-800',
        'border border-gray-200 dark:border-gray-700',
        className
      )}
    >
      {children}
    </div>
  )
}

export default Card
