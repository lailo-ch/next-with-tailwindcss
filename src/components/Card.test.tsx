import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Card from './Card'

describe('Card', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Card>
        <h1>hello world</h1>
      </Card>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
