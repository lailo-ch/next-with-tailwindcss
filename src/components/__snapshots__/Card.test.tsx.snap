// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<div
  className="rounded shadow-md p-4 sm:p-6 md:p-8 bg-white dark:bg-gray-800 border border-gray-200 dark:border-gray-700"
>
  <h1>
    hello world
  </h1>
</div>
`;
